framework/mesh/MeshGenerator/MeshGenerator_02_utils.cc

```diff
@@ -3,27 +3,54 @@
 namespace chi_mesh
 {
 
+// ###################################################################
+/**Broadcasts PIDs to other locations.*/
+void MeshGenerator::BroadcastPIDs(std::vector<int64_t>& cell_pids,
+                                  int root,
+                                  MPI_Comm communicator)
+{
+  size_t data_count = Chi::mpi.location_id == root ? cell_pids.size() : 0;
+
+  //======================================== Broadcast data_count to all
+  //                                         locations
+  MPI_Bcast(&data_count,   // buffer [IN/OUT]
+            1,             // count
+            MPI_UINT64_T,  // data type
+            root,          // root
+            communicator); // communicator
+
+  if (Chi::mpi.location_id != root) cell_pids.assign(data_count, 0);
+
+  //======================================== Broadcast partitioning to all
+  //                                         locations
+  MPI_Bcast(cell_pids.data(),             // buffer [IN/OUT]
+            static_cast<int>(data_count), // count
+            MPI_LONG_LONG_INT,            // data type
+            root,                         // root
+            communicator);                // communicator
+}
+
 // ###################################################################
 /**Determines if a cells needs to be included as a ghost or as a local cell.*/
-bool chi_mesh::MeshGenerator::CellHasLocalScope(
+bool MeshGenerator::CellHasLocalScope(
+  int location_id,
   const chi_mesh::UnpartitionedMesh::LightWeightCell& lwcell,
   uint64_t cell_global_id,
   const std::vector<std::set<uint64_t>>& vertex_subscriptions,
-  const std::vector<int64_t>& cell_partition_ids)
+  const std::vector<int64_t>& cell_partition_ids) const
```

Comment:
can you move the ") const" to the next line for readability? I finally understood that the last const applied to the whole method. the ")" threw me in!

Reply:
Our coding convention right now, in the .clang-format file, is set up to place it on the same line, which I also believe is the better way to do it because it does save space. It might be that you are not used to working with the const functions (like I was), so can I please ask you to hang in there and see if it gets better. The alternative is to go to openFOAM style and have ever increasing line counts which I don't believe you want either.